incubator/redis-auto-cluster/templates/service.yaml

```diff
@@ -0,0 +1,39 @@
+apiVersion: v1
+kind: Service
+metadata:
+{{- if .Values.serviceNameOverride }}
+  name: {{ .Values.serviceNameOverride }}
+{{- else }}
+  name: {{ template "redis-auto-cluster.fullname" . }}
+{{- end }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ template "redis-auto-cluster.name" . }}
+    chart: {{ template "redis-auto-cluster.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+  annotations:
+    service.alpha.kubernetes.io/tolerate-unready-endpoints: "false"
```

Comment:
In addition to the annotation you should also set `spec.publishNotReadyAddresses: true`.